src/read.rs

```diff
@@ -0,0 +1,67 @@
+use std::io;
+use std::io::Write;
+use std::marker::PhantomData;
+use Event;
+
+pub struct ReadHtml<'a, I> where I: Iterator<Item=Event<'a>> + 'a {
```

Comment:
I vote both, ReadHtml was just the first thing I thought of when writing it, definitely not the best name.